cast/src/lib.rs

```diff
@@ -66,9 +67,17 @@ impl Network {
     }
 }
 
-pub fn get_provider(url: &str) -> Result<JsonRpcClient<HttpTransport>> {
+pub fn get_provider(url: &str, network: &Network) -> Result<JsonRpcClient<HttpTransport>> {
     let parsed_url = Url::parse(url)?;
     let provider = JsonRpcClient::new(HttpTransport::new(parsed_url));
+
+    let rt = Runtime::new()?;
+    let provider_chain_id = rt.block_on(provider.chain_id())?;
+    let cli_chain_id = network.get_chain_id();
+    if provider_chain_id != cli_chain_id {
+        bail!("Networks mismatch: requested network is different than provider network!")
```

Comment:
Should we migrate all of `Err(anyhow!(...))` to `bail!`?

Reply:
let's. do you want me to handle this, or do you want to do this?

Reply:
I will in a separate PR after #2146 is merged